MIGRATION.md

```diff
@@ -1,296 +1,714 @@
+# Sentry JavaScript SDK Migration Docs
+
+These docs walk through how to migrate our JavaScript SDKs through different major versions.
+
+- Upgrading from [SDK 4.x to 5.x/6.x](./docs/migration/v4-to-v5_v6.md)
+- Uprading from [SDK 6.x to 7.x](./docs/migration/v6-to-v7.md)
+- Upgrading from [SDK 7.x to 8.x](./MIGRATION.md#upgrading-from-7x-to-8x)
+
 # Upgrading from 7.x to 8.x
 
-## Removal of the `MetricsAggregator` integration class and `metricsAggregatorIntegration`
+The main goal of version 8 is to improve our performance monitoring APIs, integrations API, and ESM support. This
+version is breaking because we removed deprecated APIs, restructured npm package contents, and introduced new
+dependencies on OpenTelemetry. Below we will outline the steps you need to take to tackle to deprecated methods.
 
-The SDKs now support metrics features without any additional configuration.
+Before updating to `8.x` of the SDK, we recommend upgrading to the latest version of `7.x`. You can then follow
+[these steps](./MIGRATION.md#deprecations-in-7x) remove deprecated methods in `7.x` before upgrading to `8.x`.
 
-## Updated behaviour of `tracePropagationTargets` in the browser (HTTP tracing headers & CORS)
+The v8 version of the JavaScript SDK requires a self-hosted version of Sentry TBD or higher (Will be chosen once first
+stable release of `8.x` comes out).
 
-We updated the behaviour of the SDKs when no `tracePropagationTargets` option was defined. As a reminder, you can
-provide a list of strings or RegExes that will be matched against URLs to tell the SDK, to which outgoing requests
-tracing HTTP headers should be attached to. These tracing headers are used for distributed tracing.
+## 1. Version Support changes:
 
-Previously, on the browser, when `tracePropagationTargets` were not defined, they defaulted to the following:
-`['localhost', /^\/(?!\/)/]`. This meant that all request targets to that had "localhost" in the URL, or started with a
-`/` were equipped with tracing headers. This default was chosen to prevent CORS errors in your browser applications.
-However, this default had a few flaws.
+**Node.js**: We now official support Node 14+ for our CJS package, and Node 18.8+ for our ESM package. This applies to
```

Comment:
Actually let's make this `14.8+` as this is what supports async local storage! (I'll also bump the actual engines for this)